### PR TITLE
Use new ASP.NET Core 2.1 features

### PR DESCRIPTION
**Pull request by dougbu**
- #240
- add `ModelStateInvalidFilter` to all WebHook actions
  - add `WebHookFilterProvider` class and refactor `WebHookRoutingProvider.AddFilters(...)` to there
- use `MediaTypeHeaderValue` for subtype suffix matches

nits:
- make `MediaTypeHeaderValue`s in `WebHookVerifyBodyTypeFilter` read only
- add `using`s for namespaces used only in updated doc comments
  - VS "Remove 'n Sort" doesn't remove these anymore

### CODE REVIEW

**Comment by dougbu** [comment]
@Eilon you may want to scan the increasingly-consistent doc comments I changed.
@rynowak actual product changes are pretty small in this PR (not as small as in #264 but small).

**Comment by dougbu** [comment]
:up::date: to rebase on `release/2.1`

File: src/Microsoft.AspNetCore.WebHooks.Receivers/ApplicationModels/WebHookFilterProvider.cs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            _loggerFactory = loggerFactory;
+
+            var logger = loggerFactory.CreateLogger<ModelStateInvalidFilter>();
+            _modelStateInvalidFilter = new ModelStateInvalidFilter(_behaviorOptions, logger);
+        }
+
+        /// <summary>
+        /// Gets the <see cref="IApplicationModelProvider.Order"/> value used in all
+        /// <see cref="WebHookFilterProvider"/> instances. The WebHook <see cref="IApplicationModelProvider"/> order
+        /// is
```
Comment (Eilon) [inline src/Microsoft.AspNetCore.WebHooks.Receivers/ApplicationModels/WebHookFilterProvider.cs]: `is:`

File: src/Microsoft.AspNetCore.WebHooks.Receivers/ApplicationModels/WebHookMetadataProvider.cs [hunk 2]
```diff
@@ -100,20 +102,25 @@ public class WebHookMetadataProvider : IApplicationModelProvider
 
         /// <summary>
         /// Gets the <see cref="IApplicationModelProvider.Order"/> value used in all
-        /// <see cref="WebHookMetadataProvider"/> instances. The recommended <see cref="IApplicationModelProvider"/>
-        /// order is
+        /// <see cref="WebHookMetadataProvider"/> instances. The WebHook <see cref="IApplicationModelProvider"/> order
+        /// is
```
Comment (Eilon) [inline src/Microsoft.AspNetCore.WebHooks.Receivers/ApplicationModels/WebHookMetadataProvider.cs]: `:`

File: src/Microsoft.AspNetCore.WebHooks.Receivers/ApplicationModels/WebHookModelBindingProvider.cs [hunk 3]
```diff
@@ -37,24 +37,29 @@ public WebHookModelBindingProvider(ILoggerFactory loggerFactory)
 
         /// <summary>
         /// Gets the <see cref="IApplicationModelProvider.Order"/> value used in all
-        /// <see cref="WebHookModelBindingProvider"/> instances. The recommended
-        /// <see cref="IApplicationModelProvider"/> order is
+        /// <see cref="WebHookModelBindingProvider"/> instances. The WebHook <see cref="IApplicationModelProvider"/>
+        /// order is
```
Comment (Eilon) [inline src/Microsoft.AspNetCore.WebHooks.Receivers/ApplicationModels/WebHookModelBindingProvider.cs]: `:`

File: src/Microsoft.AspNetCore.WebHooks.Receivers/ApplicationModels/WebHookRoutingProvider.cs [hunk 4]
```diff
@@ -8,57 +8,58 @@
...
             _eventMapperConstraint = eventMapperConstraint;
-            _loggerFactory = loggerFactory;
         }
 
         /// <summary>
         /// Gets the <see cref="IApplicationModelProvider.Order"/> value used in all
-        /// <see cref="WebHookRoutingProvider"/> instances. The recommended <see cref="IApplicationModelProvider"/>
-        /// order is
+        /// <see cref="WebHookRoutingProvider"/> instances. The WebHook <see cref="IApplicationModelProvider"/> order
+        /// is
```
Comment (Eilon) [inline src/Microsoft.AspNetCore.WebHooks.Receivers/ApplicationModels/WebHookRoutingProvider.cs]: `:`

**Comment by dougbu** [comment]
ec834ebabb

**Pull request closed without merging**